lib/Net/Google/Drive/Simple.pm

```diff
@@ -721,23 +717,48 @@ sub file_metadata {
 
     my $url = URI->new( $self->{api_file_url} . "/$file_id" );
 
-    my $req = &HTTP::Request::Common::GET(
-        $url->as_string,
-        $self->{oauth}->authorization_headers(),
-    );
+    return $self->http_json($url);
+}
 
-    my $ua   = LWP::UserAgent->new();
-    my $resp = $ua->request($req);
+###########################################
+sub _content_sub {
+###########################################
+    my $filename  = shift;
+    my @stat      = stat $filename;
+    my $remaining = $stat[7];
+    my $blksize   = $stat[11] || 4096;
 
-    if ( $resp->is_error ) {
-        $self->error( $resp->message() );
-        return;
-    }
+    die "$filename not a readable file with fixed size"
+      unless -r $filename
+          and $remaining;
 
-    my $data = from_json( $resp->content() );
+    my $fh = IO::File->new($filename, 'r')
+      or die "Could not open $filename: $!";
+    $fh->binmode;
 
-    # return $self->data_factory( $data );
-    return $data;
+    return sub {
+        my $buffer;
+
+        # upon retries the file is closed and we must reopen it
+        unless ($fh->opened) {
+            $fh = IO::File->new($filename, 'r')
+              or die "Could not open $filename: $!";
+            $fh->binmode;
+            $remaining = $stat[7];
```

Comment:
maybe we could avoid that duplicate logic and only do it once not defined or unopened?